### PR TITLE
[coreSNTP] CBMC Proof for Sntp_CalculatePolIInterval Function

### PR DESCRIPTION
**Pull request by gshvang**
This PR contains CBMC Proof for Sntp_CalculatePolIInterval Function.

### CODE REVIEW

File: test/cbmc/proofs/Sntp_CalculatePollInterval/Makefile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+HARNESS_ENTRY = harness
+HARNESS_FILE = Sntp_CalculatePollInterval_harness
+
+# This should be a unique identifier for this proof, and will appear on the
+# Litani dashboard. It can be human-readable and contain spaces if you wish.
+PROOF_UID = Sntp_CalculatePollInterval
+
+MAX_BOUND_FOR_LOOP=33
```
Comment (yourslab) [inline test/cbmc/proofs/Sntp_CalculatePollInterval/Makefile:11]: Is 33 necessary? Does 32 work?

File: test/cbmc/proofs/Sntp_CalculatePollInterval/Sntp_CalculatePollInterval_harness.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+void harness()
+{
+    uint16_t clockFreqTolerance;
+    uint16_t desiredAccuracy;
+    uint32_t pollInterval;
+    SntpStatus_t sntpStatus;
+
+    sntpStatus = Sntp_CalculatePollInterval( clockFreqTolerance, desiredAccuracy, &pollInterval );
+
+    __CPROVER_assert( ( sntpStatus == SntpErrorBadParameter || sntpStatus == SntpSuccess ), "The return value is a valid SNTP status." );
```
Comment (aggarw13) [inline test/cbmc/proofs/Sntp_CalculatePollInterval/Sntp_CalculatePollInterval_harness.c:39]: Is the message string printed when the assert is satisfied? (Usually, asserts functions carry message strings for printing when expression is not satisfied..)

File: test/cbmc/proofs/Sntp_CalculatePollInterval/Makefile (hunk 1 above)
Comment (aggarw13) [inline test/cbmc/proofs/Sntp_CalculatePollInterval/Makefile:11]: A comment about the rationale behind the loop iteration value will be helpful

File: test/cbmc/proofs/Sntp_CalculatePollInterval/Sntp_CalculatePollInterval_harness.c (hunk 2 above)
Comment (gshvang) [reply test/cbmc/proofs/Sntp_CalculatePollInterval/Sntp_CalculatePollInterval_harness.c:39]: Yeah, assert message is printed when the message strings are. not satisfied. I Will update the message.

File: test/cbmc/proofs/Sntp_CalculatePollInterval/Makefile (hunk 1 above)
Comment (gshvang) [reply test/cbmc/proofs/Sntp_CalculatePollInterval/Makefile:11]: We set the unwinding loop bound to be 1 more than the actual iterations it take, to cover the case when the loop exits. So 32 iterations will set the value to 1 , to test the condition when loop exits we need to iterate once more.

**Review by aggarw13 — APPROVED** [review]

**Review by yourslab — APPROVED** [review]

**Comment by gshvang** [comment]
Sntp_CalculatePollInterval_Proof_report

**Review by n9wxu — APPROVED** [review]

**Pull request merged**